tests/unit/test__datastore_query.py

```diff
@@ -152,8 +152,8 @@ def test_count_by_skipping_w_a_result(run_query):
                     more_results=_datastore_query.NOT_FINISHED,
                     skipped_results=1000,
                     entity_results=[],
-                    end_cursor=b"dontlookatme",
```

Comment:
Looking at this change it seemed reasonable, but I am a bit surprised to see test changes. Can you explain why?

Reply:
Short version: the buggy behavior of #575 & #584 was being defended by tests.

Extended: in this PR I am basically restoring the behavior before #528 for the non-emulator case so I am more or less reverting the test changes of that PR as well.

More extended: these local line changes here and the other 1-2 line changes are there because of the following behavior was changed: between #528 and this PR, if an empty batch came from the datastore, then it was assumed that it's a datastore emulator and therefore `skipped_cursor` was used instead of `end_cursor`. I don't think that this is a good behavior and always the `end_cursor` was used before #528 so I think this should be a safe change. 

And of course `test_count_by_skipping_emulator` had to be rewritten because the emulator behavior was changed entirely.